src/App.tsx

```diff
@@ -8,51 +8,158 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import * as postService from './api/users';
+import { getPostsByUserId } from './api/posts';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import * as commentService from './api/comments';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const [posts, setPosts] = useState<Post[] | []>([]);
+  const [loadingPosts, setLoadingPosts] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [postsErrorMessage, setPostsErrorMessage] = useState('');
+
+  const [comments, setComments] = useState<Comment[] | []>([]);
+  const [loadingComments, setLoadingComments] = useState(false);
+  const [commentsErrorMessage, setCommentsErrorMessage] = useState('');
+
+  const [isNewCommentFormOpened, setIsNewCommentFormOpened] = useState(false);
+
+  const hasNoPosts =
+    !loadingPosts && selectedUser && !posts.length && !postsErrorMessage;
+
+  function getNewCommentId(currentComments: Comment[]) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const getNewCommentId = (currentComments: Comment[]) => {
```